brainscore/benchmarks/moeller2017.py

```diff
@@ -0,0 +1,503 @@
+        '''
+        candidate.start_recording()
+        recordings = candidate.look_at(self._stimulus_set_training)
+        samples = 500
+
+        stimulus_set = truth = []
+        for object_name in set(recordings.object_name):
+            recordings, conditions = self._sample_recordings(recordings.sel(object_name=object_name),
+                                                             samples=samples)
+            stimulus_set.append(recordings)
+            truth += np.array(conditions) == 'same_id'
+        stimulus_set = np.vstack(stimulus_set)
+
+        self._decoder = LinearRegression()
+        self._decoder.fit(stimulus_set, truth)
+
+    def _sample_recordings(self, category_pool, samples=500):
+        '''
+        Create an array of randomly sampled recordings, each line is one task, i.e. two recordings which are to be
+        judged same vs. different ID
+        :param category_pool: DataArray: Model IT recordings, assumed be from one category only
+        :param samples: int: number of samples
+        :return: array (samples x length of two concatenated recordings), each line contains two recordings
+                 ground truth, for each line in array, specifying if the two recordings belong to the same/different ID
+        '''
+        rng = np.random.default_rng(seed=self._seed)
+        recording_size = len(category_pool[0].values)
+        random_indeces = rng.integers(0, len(category_pool), (samples, 2))
+
+        sampled_recordings = np.full((samples, recording_size * 2), np.nan)
+        for i, (random_idx_same, random_idx_different) in enumerate(random_indeces):
+            # 'same_id': object_id ==, image_id!=
+            image_one_same = category_pool[random_idx_same]
+            sampled_recordings[i, :recording_size] = image_one_same.values
+            sampled_recordings[i, recording_size:] = rng.choice(category_pool.where(
+                category_pool.object_id == image_one_same.object_id and
+                category_pool.image_id != image_one_same.image_id))
+
+            # 'different_id': object_id !=
+            image_one_diff = category_pool[random_idx_different]
+            sampled_recordings[i + samples, :recording_size] = image_one_diff.values
+            sampled_recordings[i + samples, recording_size:] = rng.choice(category_pool.where(
+                category_pool.object_id != image_one_diff.object_id))
+
+        conditions = ['same_id'] * samples + ['different_id'] * samples
+        return sampled_recordings, conditions
+
+    def _compute_perturbation_coordinates(self, candidate):
+        '''
+        Save stimulation coordinates (x,y) to self._perturbation_coordinates
+        :param candidate: BrainModel
+        '''
+        candidate.start_recording('IT', time_bins=[(50, 100)])
+        recordings = candidate.look_at(self._stimulus_set_FP)  # vs _training_assembly
+
+        # 1 smooth spatial pattern by smoothing activity with gaussian kernel 1mm

```

Comment:
this is effectively converting electrode array recordings into fMRI correct? If yes, then I think this should also go into the model, i.e. `candidate.start_recording('IT', technique='fMRI', ...)`